newIDE/app/src/ObjectsRendering/PixiResourcesLoader.js

```diff
@@ -146,6 +147,133 @@ export default class PixiResourcesLoader {
     return loadedTextures[resourceName];
   }
 
+  static createTileSetResource(
```

Comment:
I have yet to review this, but I'm wondering if we could be able to move this into the JsExtension.js later.

Reply:
I think the reasoning behind this was to follow the way loadedTextures are being cached and reused as a type of a resource.
The same thing happens for tilesets - a tileset is created from a texture+ a json resource. If a tileset with the same texture and json resource already exists in the cache - its reused.

Should I perhaps move this stuff to its own file in the same folder? PixiResourcesLoader.js seemed like the right  place to me, since this is a resource to be consumed by PIXI. Also the tileset may be used by just the Tilemap here, but later on we could use tilesets to reduce draw calls in general by packing and even give them a better way of being previewed.

Of course it does make more sence to be a part of the extension.js file instead, but it will then break away from the current pattern and no longer be treated as a type of a global resource. I can just keep the cached tilesets in the extension instead

Reply:
> I noticed that I might have to write the same tileset creation code again for the runtime, which seems a bit wasteful. Would having it in the extension perhaps make things easier?

True, if we're reusing code, either we:
- copy/paste it, or write it twice.
- or we make a "UMD library" for it :) So that it can be loaded in Electron and in the browser.

Do as what's easiest for you. The mini "UMD library" would be awesome as it would allow the same file to be used both at runtime and in Electron - but we can do that later too if it's too complicated for now. We'll have to allow this file to be loaded in `ObjectsRenderingService.js` like we did for PIXI.

> Should I perhaps move this stuff to its own file in the same folder?

Yes ideally. Not a big deal for now, if you wish you can continue without doing this refactoring, but I'll still prefer us to have this done before we merge it. 
While it's true that tileset (texture + json tiled file) can be seen as a resource, it is not one for now: the only primitive resources are audio files, texture files, video files, font files and json files.
I would prefer not to muddy the waters in the existing `PixiResourcesLoader` and the equivalent in the runtime. 
The tileset are some kind of "composite" resource, composed of a texture and a json. Let's for now keep them in a cache the extension and we can refactor this later when we have a proven use case of reusing them.